akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala

```diff
@@ -848,11 +851,15 @@ object Balance {
  *
  * '''Completes when''' upstream completes
  *
- * '''Cancels when''' all downstreams cancel
+ * '''Cancels when''' If eagerCancel is enabled: when any downstream cancels; otherwise: when all downstreams cancel
  */
-final class Balance[T](val outputPorts: Int, val waitForAllDownstreams: Boolean) extends GraphStage[UniformFanOutShape[T, T]] {
+final class Balance[T](val outputPorts: Int, val waitForAllDownstreams: Boolean, val eagerCancel: Boolean) extends GraphStage[UniformFanOutShape[T, T]] {
   // one output might seem counter intuitive but saves us from special handling in other places
   require(outputPorts >= 1, "A Balance must have one or more output ports")
+
+  @deprecated("Use the constructor which also specifies the `eagerCancel` parameter")
```

Comment:
this should include the version number, so 2.5.12 if we can get it in before we release that

Reply:
@Hawstein ,with Java's `@Deprecated ` and with `since 2.5.12` info.

Reply:
Done. @johanandren @hepin1989